ReactSkia/sdk/OnScreenKeyBoard.cpp

```diff
@@ -14,606 +14,609 @@
+        return oskHandle;
+    }
 
-    OnScreenKeyboard &oskHandle=OnScreenKeyboard::getInstance();
+    OSKErrorCode OnScreenKeyboard::launch(OSKConfig oskConfig) {
+        RNS_LOG_TODO("Need to do emit , keyboardWillShow Event to APP");
 
-    if((oskHandle.oskState_ == OSK_STATE_LAUNCH_INPROGRESS) || (oskHandle.oskState_ == OSK_STATE_ACTIVE))
-        return OSK_ERROR_ANOTHER_INSTANCE_ACTIVE;
+        OnScreenKeyboard &oskHandle=OnScreenKeyboard::getInstance();
 
-    std::scoped_lock lock(oskLaunchExitCtrlMutex);
+        if((oskHandle.oskState_ == OSK_STATE_LAUNCH_INPROGRESS) || (oskHandle.oskState_ == OSK_STATE_ACTIVE))
+            return OSK_ERROR_ANOTHER_INSTANCE_ACTIVE;
 
-    oskHandle.oskState_=OSK_STATE_LAUNCH_INPROGRESS;
-    oskHandle.launchOSKWindow(oskConfig);
-    return OSK_LAUNCH_SUCCESS;
-}
+        std::scoped_lock lock(oskLaunchExitCtrlMutex);
 
-void OnScreenKeyboard::exit() {
-    RNS_LOG_TODO("Need to emitkeyboardWillHide Event  to APP");
+        oskHandle.oskState_=OSK_STATE_LAUNCH_INPROGRESS;
+        oskHandle.launchOSKWindow(oskConfig);
+        return OSK_LAUNCH_SUCCESS;
+    }
 
-    OnScreenKeyboard &oskHandle=OnScreenKeyboard::getInstance();
+    void OnScreenKeyboard::exit() {
+        RNS_LOG_TODO("Need to emitkeyboardWillHide Event  to APP");
 
-    if((oskHandle.oskState_ == OSK_STATE_EXIT_INPROGRESS) || (oskHandle.oskState_ == OSK_STATE_INACTIVE))
-        return;
+        OnScreenKeyboard &oskHandle=OnScreenKeyboard::getInstance();
 
-    oskHandle.oskState_=OSK_STATE_EXIT_INPROGRESS;
-/* Stop Listening for Hw Key Event*/
-    if(oskHandle.oskEventId_) {
-        NotificationCenter::OSKCenter().removeListener(oskHandle.oskEventId_);
-        oskHandle.oskEventId_= -1;
-    }
-    std::scoped_lock lock(oskLaunchExitCtrlMutex);
-    oskHandle.closeWindow();
-    oskHandle.oskState_=OSK_STATE_INACTIVE;
-}
-
-void  OnScreenKeyboard::updatePlaceHolderString(std::string PHdisplayString) {
-    RNS_LOG_TODO("updatePlaceHolderString:Implementation yet to be done");
-    return;
-}
-
-void OnScreenKeyboard::launchOSKWindow(OSKConfig oskConfig) {
-
-    memcpy(&oskConfig_,&oskConfig,sizeof(oskConfig));
-
-    SkSize mainscreenSize_=RnsShell::PlatformDisplay::sharedDisplay().screenSize();
-    if(screenSize_ != mainscreenSize_) {
-        generateOSKLayout_=true;
-        screenSize_=mainscreenSize_;
-    } else {
-        generateOSKLayout_=false;
-    }
-    //Set up OSk configuration
-    if(oskConfig_.theme == OSK_LIGHT_THEME) {
-        bgColor_ = OSK_LIGHT_THEME_BACKGROUND_COLOR ;
-        fontColor_ = OSK_LIGHT_THEME_FONT_COLOR;
-    } else {
-        bgColor_ = OSK_DARK_THEME_BACKGROUND_COLOR ;
-        fontColor_ =OSK_DARK_THEME_FONT_COLOR;
+        if((oskHandle.oskState_ == OSK_STATE_EXIT_INPROGRESS) || (oskHandle.oskState_ == OSK_STATE_INACTIVE))
+            return;
+
+        oskHandle.oskState_=OSK_STATE_EXIT_INPROGRESS;
+    /* Stop Listening for Hw Key Event*/
+        if(oskHandle.oskEventId_) {
```

Comment:
This check must be > 0 right. We are setting to -1 means this check will always pass..

Reply:
Yes. Wil update